fred2/ignoreordersdlg.cpp

```diff
@@ -101,53 +87,24 @@ BOOL ignore_orders_dlg::OnInitDialog()
 
 	}
 
-	// set the checkboxes for the orders accepted
-	m_num_checks_active = 0;
-	for (size_t order_id : default_orders){
-		// Not enough space to display checkboxes for all comm orders!
-		// Need to add more checkboxes.
-		if (m_num_checks_active >= MAX_CHECKBOXES)
-		{
-			Warning(LOCATION, "Tried to add more orders than FRED2 has checkboxes for them (can be max %d)!", MAX_CHECKBOXES);
-			break;
-		}
-
-		check_boxes[m_num_checks_active].button->SetWindowText(Player_orders[order_id].localized_name.c_str());
-		check_boxes[m_num_checks_active].id = (int)order_id;
-		m_num_checks_active++;
+	CCheckListBox* list = ((CCheckListBox*)GetDlgItem(IDC_IGNORE_ORDERS_LIST));
 	
-	}
-
-	// resize the dialog by shrinking the height by the number of checkboxes that
-	// we removed
-	GetWindowRect( &window_size );
-
-	// hide the rest of the dialog items
-	last_bottom = 0;
-	for ( i = MAX_CHECKBOXES - 1; i >= m_num_checks_active; i-- ) {
-		RECT check_size;
 
-		// get the size of the checkbox, then hide it.
-		check_boxes[i].button->GetWindowRect( &check_size );
-		check_boxes[i].button->ShowWindow(SW_HIDE);
-
-		// shrink the size of the parent window by the size of the checkbox
-		if ( last_bottom != 0 )
-			window_size.bottom -= (last_bottom - check_size.bottom );
+	for (size_t order_id : default_orders){
+		int checkboxNr = (int) m_orderList.size();
 
-		last_bottom = check_size.bottom;
+		list->AddString(Player_orders[order_id].localized_name.c_str());
+		list->SetCheck(checkboxNr, BST_UNCHECKED);
```

Comment:
Isn't this index out of range?

Also using `Nr` to mean "number" seems unintuitive to me. That convention isn't used elsewhere in the codebase.